### DISCUSSION TITLE
Font size is way too big!?!?

### DISCUSSION THREAD

**Discussion by polloze** [NightscoutFoundation/xDrip#3782] [Bug reports]
Hello!

Does anybody know what causes this issue as you can see in the picture??

My Pixel 7 broke so I had to switch to a new phone. It's a Pixel 9 with Android 15 and all available updates to date!

I exported my xDrip settings on the Pixel 7, installed xDrip on the Pixel 9 and after the first start of the app I imported those settings and everything right away.

xDrip itself is the newest stable release version as you can see from the status page picture.

Navid already told me to have aook in the Pixel 9 settings menu to change the font size. It's already the second lowest I can choose. If I choose th smallest font size it clearly changes the font size in xDrip. Unfortunately everything else just gets to small for my eyes why this isn't a solution, regardless it's still bigger than it supposed to be.

Any ideas??![Screenshot_20241123-122551.png](https://github.com/user-attachments/assets/d174a9ed-27da-4ee1-b437-be9d22c55644)

![Screenshot_20241123-122539.png](https://github.com/user-attachments/assets/8fc7b69b-3f62-4402-8ea4-c14e6032268a)

![Screenshot_20241123-122543.png](https://github.com/user-attachments/assets/f19f216c-8213-4843-aea9-dbc2c8a04947)

**Comment by parapenT1sta** [comment]
Same thing with Samsung Z Fold 4, the inner screen get status line font too big.

https://github.com/NightscoutFoundation/xDrip/discussions/3196

**Reply by Navid200** [reply]
I don't see how what you have is related to the problem described here.  You can clearly see that the font size here is wrong.

**Comment by Navid200** [comment]
@polloze  You said that you transferred settings from another phone.
But, you never explained if everything was OK on the new phone before transferring the settings.

**Reply by polloze** [reply]
@Navid200 Well, I didn't pay attention after installing xDrip. I wanted to reinstall xDrip to check this. I just didn't have enough time at the weekend. Hopefully I can check this within the next few days.

**Comment by vincepf** [comment]
I have this same concern after switching to a new phone with LineageOS Android 15. Anyone find a fix? I've tried system font size. Can't find a font size setting in xDrip.

**Reply by Navid200** [reply]
I don't know how this is possible.  I don't believe there is any setting in xDrip that allows changing a font size.
If there is a description how to recreate this problem, I may be able to help.

**Reply by polloze** [reply]
I wouldnt be surprised if it's an issue of Android 15 and Xdrip... I've had another app that had kinda the same issue. And they fixed it with some changes to the code.

**Reply by Navid200** [reply]
There is nothing I can do if I cannot see the problem.
I am running Android 15.

**Reply by polloze** [reply]
I understand Navid. There's just gotta be some reason why some people do have this issue. Just curious what phone you use?

I've seen a post here about this issue too. There's a bunch of people facing this problem. Well, to me it's not really a problem. Just something that looks odd.

**Reply by Navid200** [reply]
My only phone that runs Android 15, my main phone, is Pixel 6a.

**Reply by vincepf** [reply]
@Navid200 Thank you for the reply. I'm sure there are more pressing issues to be dealt with, so don't feel the need to spend any real time working on this one. I appreciate all the time you spend and effort you give for us xDrip users.

Mine is a Pixel 4XL with LineageOS 22.1 installed. 

The way the problem appeared: installed xDrip. Open xDrip and the font is too big.

Again, as @polloze mentioned, it does not affect the functionality of the app. Just looks odd.

![Screenshot_20250404-064423_xDrip+](https://github.com/user-attachments/assets/8275e9a2-1d9f-4cb2-ab4f-6351ab2c3fcb)

**Comment by vincepf** [comment]
I resolved mine by changing the "Display Size" to the next bigger setting. xDrip looks normal when I'm on the setting pictured. Any display size setting below that makes the status line text huge.
![Screenshot_20250412-064326_Settings](https://github.com/user-attachments/assets/d9eaa45a-795a-4a1d-a4bf-dbff0c181040)

**Comment by mcurly** [comment]
Hi, same issue here. 

![Screenshot_20250416-042613_Settings](https://github.com/user-attachments/assets/4d7698db-d4cd-47d6-ad05-85c9d57c532a)
![Screenshot_20250416-042655_xDrip+](https://github.com/user-attachments/assets/ae1b0601-abdb-4c1f-8f56-e3deffe9a918)

**Reply by mcurly** [reply]
It can be workaround with making the display fonts look bigger: 

![Screenshot_20250416-043214_Settings](https://github.com/user-attachments/assets/488de535-6443-4ed8-8d78-79f6dfc51b8d)
![Screenshot_20250416-043217_xDrip+](https://github.com/user-attachments/assets/2058d222-13f3-4b3a-8a86-d965941903a9)

**Comment by kenkolden** [comment]
just saw this issue with fonts too large. Open Android settings select "Display size and text" at the bottom select "Reset settings"

**Reply by polloze** [reply]
Unfortunately that doesnt work for me.

**Comment by Navid200** [comment]
Does the phone of those of you who experience this problem have one screen or two different screens with different resolutions?

**Reply by polloze** [reply]
Hello Navid, I have just one screen. Never had anymore than one!

**Reply by Navid200** [reply]
I set up Pixel 9 in Android Studio and I don't see the large fonts that you see.
There are others who use Pixel 9.  I'm not sure why you experience this.

**Reply by polloze** [reply]
What do you mean by saying "set up Pixel 9 in Android Studio"??
If you dont know than I definitely dont know why me (and some other people) experience large fonts. ;)

**Reply by Navid200** [reply]
You can create a virtual machine in Android studio and run xDrip on it.

xDrip checks the screen size of the device and based on that figures out if the device is a phone or a tablet.
If it is a tablet, it increases the font size.  

There are many people who use standard Pixel 9 and if every one had this issue, we would have heard from all of them.
I suspect something on your phone makes xDrip think it is a tablet.  

Is there anything on your phone that is not standard?  Have you rooted the phone?

**Reply by polloze** [reply]
Oh ok, I might gonna look into it how to create a VM in Android Studio.
What you describe makes sense...

I bet there would be more people complaining about this issue.
Not standard? Not rooted for sure! It's basically stock. I got AdAway installed as a VPN service. But other than that there's nothing in particular that could force such a behavior in my opinion...

**Comment by vincepf** [comment]
This behavior is the same on my Pixel 4XL with LineageOS 22.2 as well as my new Pixel 6a. Neither has a separate screen. Both resolve when setting the display size one higher than the smallest, as described in previous posts.

**Reply by Navid200** [reply]
I have a Pixel 6a and have never experienced this.

**Reply by vincepf** [reply]
My son has a Pixel 6a and his behaves identical to mine. Change the "display size" in phone settings to the smallest and the status line font becomes oversized.

**Reply by Navid200** [reply]
OK, I see that.  I thought it was the display font size that was causing it.  But, I can see it happening setting display size too small.

**Reply by Navid200** [reply]
A long time ago (2016), a change was made to xDrip to enlarge the font size on tablets:
https://github.com/NightscoutFoundation/xDrip/pull/29 

It categorizes a device based on its screen size into 4 categories, small, normal, large and extra large.
Only for large and extra large, the main glucose reading font size and extra status line font size is enlarged.

Today, there are so many different screen sizes and some of them, as you have experienced, are categorized as large.  I just verified that on my Pixel 6a, if I set the screen size to minim, xDrip categorizes it as a large screen.  That is why the extra status line font size is increased.

If we don't do that and you change the screen size to minimum, the extra status line fonts become impossible to see.

Let me see what I can do about this.

**Reply by Navid200** [reply]
@jamorham @tzachi-dar 

I see two possible approaches:

1- Completely remove the automatic font size adjustment from xDrip.  Add a font size adjustment for major components on screen to the display settings so that the user can adjust the font size if needed.

2- Increase the number of categories for the existing automatic font size adjustment for example from 4 to 8 so that there would be more resolution.

Either one of those has pros and cons.
I am going to work on 2 to see if I can come up with a submission that may have a chance of being approved.

**Comment by Navid200** [comment]
This is a test release that I have created that should fic this.  You can install it to run in parallel with your official xDrip.  So, there is no need to uninstall anything.  Install and see if it is any better.  Please report the results.  

Thanks

https://drive.google.com/file/d/18oDZQn3oMWA-W-tbqDXOBJX_wU1I_R6Y/view?usp=sharing

**Reply by polloze** [reply]
Wooo, first of all I wanna say thank you that this issue get's it's attention finally. :-) I didnt really bother me at all. It'd be neat to see the font size as it supposed to be...

So I did install your test file and got this screenshot. Hopefully it does help you!?
<img width="1080" height="2424" alt="Screenshot_20250815-095109" src="https://github.com/user-attachments/assets/33ae0cb2-b81d-4724-b4e7-15a8650a99a4" />

**Reply by Navid200** [reply]
OK, this is my fault.
Would you please set up your existing xDrip as a master, either Nightscout or xDrip Sync?  Then, use this xDrip_Test install as a follower.
Set it up to show the extra status line.
Compare the font size of the extra status line between the two xDrips, which is the issue complained about in this discussion.  

Thanks

**Reply by polloze** [reply]
I thougt it might not be enough just like that. 8)

Finally I got it setup to get the data also at the xDrip_Test... and yes, it loosk much better. :)
<img width="1080" height="2424" alt="Screenshot_20250815-141709" src="https://github.com/user-attachments/assets/aafc5ff4-6274-4253-8b54-9dbe23e4cadc" />

**Reply by Navid200** [reply]
Thanks

**Reply by polloze** [reply]
Navid, will this already be fixed with the next stable release?

**Reply by Navid200** [reply]
No,

I have opened a submission that adds a setting that lets you disable font size adjustments by xDrip:
https://github.com/NightscoutFoundation/xDrip/pull/4128

It has not been reviewed yet.

**Reply by polloze** [reply]
I see. The only thing matters is that it get's the attention to be fixed at some point in the future. Thank you.

**Comment by Navid200** [comment]
As of the 2025.09.19 release, there is a new settings t:
Settings > Display Settings > Font Settings > Enlarge fonts on larger displays

If you disable this setting, xDrip will not automatically enlarge the fonts.  This should reduce the font size for you.

**Reply by polloze** [reply]
> As of the 2025.09.19 release

This is a stable release??

**Reply by Navid200** [reply]
No, There are 2 stable releases a year.  This is the latest Nightly release.
In about 6 months to a year from now, this will be in a stable release.

**Reply by polloze** [reply]
Uhm, I just downloaded the nightly. To my surprise there is no "Font Settings". Any ideas!?

**Reply by parapenT1sta** [reply]
I downloaded it and there is the option. Check in Display Settings.

**Reply by polloze** [reply]
@parapenT1sta That's good for you. Telling me to check it again doesn't help because there is no font settings for some reason!

**Reply by polloze** [reply]
@parapenT1sta Ok. I have to admit that I thought I've updated to the latest nightly... I didn't. I was so busy today with a lot of different things that I didn't pay attention. Sorry for my reaction!

**Comment by mcurly** [comment]
Hello. Thank you.

Heberto Cristiano Machado

***@***.***

A sexta, 19/09/2025, 12:04, Navid ***@***.***> escreveu:

**Comment by Navid200** [comment]
@polloze You can check the classic status page to see which version you are running.

**Reply by polloze** [reply]
I just came back home and did check it again. And then I just realized that I was wrong earlier today when I thought I run the latest nightly. After running an update, installing the latest nightly I do see now the font settings. Sorry!
